src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/RequestExecutionQueue.cs

```diff
@@ -206,7 +206,7 @@ private async Task ProcessQueueAsync()
                         // though these errors don't put us into a bad state as far as the rest of the queue goes.
                         // Furthermore we use Task.Run here to protect ourselves against synchronous execution of work
                         // blocking the request queue for longer periods of time (it enforces parallelizabilty).
-                        _ = Task.Run(() => HandleNonMutatingRequestErrorAsync(work.StartRequestAsync(context, cancellationToken)), cancellationToken);
+                        _ = WrapStartRequestTaskAsync(Task.Run(() => work.StartRequestAsync(context, cancellationToken), cancellationToken));
```

Comment:
My suggestion on naming was on the contigency that it would be called for both `StartRequest` calls.. not sure the name makes sense if it only allows wrapping of non-mutating requests. Rename? or add a `bool isMutating` or `bool isBlocking` parameter?